### ISSUE TITLE
Consider adding `use_super_parameters`

### ISSUE THREAD

**Issue by mit-mit** [dart-lang/lints#75]
**Describe the rule you'd like to see added and to what rule set**
Dart 2.17 adds the new super params feature (see additional info for an example), and in https://github.com/flutter/flutter/issues/100575 we're migrating Flutter to use.

I propose we add the `use_super_parameters` lint, which enforces this style, to recommended.

**Additional context**

Without super parameters:
```dart
class MyWidget extends StatelessWidget {
  const MyWidget({ Key? key }) : super(key: key);
}
```

With super parameters:
```dart
class MyWidget extends StatelessWidget {
  const MyWidget({ super.key });
}
```

**Comment by goderbauer** [comment]
We should make sure that the IDE macros are updated to produce valid code with this lint enabled before making it a recommended one. Work for that is tracked here:
* https://github.com/Dart-Code/Dart-Code/issues/3899
* https://github.com/flutter/flutter-intellij/issues/6071

**Comment by mit-mit** [comment]
WDYT @pq ?

**Comment by pq** [comment]
I'm in favor.  I think it does a good job of nudging folks towards using `super`.  There are a few open issues but luckily they're [false-negatives](https://github.com/dart-lang/linter/issues?q=is%3Aissue+use_super_parameters+is%3Aopen+label%3A%22false+negative%22).

**Comment by bwilkerson** [comment]
I'll just note that false negatives are also somewhat problematic because fixing them is technically a breaking change. The more users of a lint the bigger the impact of the breaking change.

We might want to wait to add the lint until we can address the false negatives (depending on the nature of the false positives).

**Comment by pq** [comment]
Thanks Brian.  Agreed!

> We might want to wait to add the lint until we can address the false negatives (depending on the nature of the false positives).

If we can prioritize the work, this would be my preference.

**Comment by devoncarew** [comment]
This has been approved for the `recommended` set.

**Comment by parlough** [comment]
Great to hear! If this is graduating to the recommended set, it might be worth increasing the priority on https://github.com/dart-lang/sdk/issues/51578.
